src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -1220,11 +1220,13 @@ private fun checkTypesAreSized(holder: RsAnnotationHolder, fn: RsFunction) {
     }
 }
 
-private fun RsAttr.isBuiltinWithName(target: String): Boolean {
+private fun RsAttr.isBuiltinAttrWithName(target: String): Boolean {
     val name = metaItem.name ?: return false
 
     if (name != target) return false
     if (name !in STD_ATTRIBUTES) return false
 
-    return !hasInScope(name, MACROS)
+    val fn = findInScope(name, MACROS) as? RsFunction
+
+    return !(fn != null && fn.isAttributeProcMacroDef)
```

Comment:
Consider to use something like
```suggestion
    val fn = findInScope(name, MACROS) as? RsFunction ?: return true

    return !fn.isAttributeProcMacroDef
```
or
```suggestion
    val fn = findInScope(name, MACROS) as? RsFunction

    return fn?.isAttributeProcMacroDef != true
```